sensor/tests/test_location.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+from django.urls import reverse
+from django.test import TestCase, Client
+from rest_framework import status
+
+from sensor.models import *
+from .context import TestContext
+from api_key.models import ApiKey
+
+
+class LocationTest(TestCase):
+
+    def setUp(self):
+        self.context = TestContext()
+        self.client = Client()
+        self.url = '/sensor/api/location/create'
+        self.dev_id = self.context.dev.id
+        self.key = str(self.context.dev.post_key.external_key)
+        self.name, self.lat, self.lon, self.alt = 'Planet Hoth', 60.588, 7.48, 1222
+
+    def test_get_location(self):
+        device = Device.objects.get(pk=self.context.dev.id)
+        # name = "Ulriken", latitude = 200, longitude = 120, altitude = 600
+        self.assertEqual('Ulriken', device.location.name)
+        self.assertEqual(200, device.location.latitude)
+        self.assertEqual(120, device.location.longitude)
+        self.assertEqual(600, device.location.altitude)
+
+    def _assert_error_msg(self, err, msg):
+        """Asserts msg in err.message"""
+        self.assertIn(msg, err.message)
+
+    def test_create_location_key_issues(self):
+        with self.assertRaises(KeyError) as ctx:
+            response = self.client.get("%s/%s/" % (self.url, self.dev_id))
+        self._assert_error_msg(ctx.exception, 'Missing API-key')
+
+        with self.assertRaises(KeyError) as ctx:
+            args = (self.url, self.dev_id, 'no-such-key')
+            response = self.client.get("%s/%s/?key=%s" % args)
+        self._assert_error_msg(ctx.exception, 'Invalid')
+
+        other_existing_key = ApiKey.objects.create(description="Newerkey")
+        with self.assertRaises(KeyError) as ctx:
+            args = (self.url, self.dev_id, other_existing_key.external_key)
+            response = self.client.get("%s/%s/?key=%s" % args)
+        self._assert_error_msg(ctx.exception, 'Invalid')
+
+
+    def test_create_location_missing_fields(self):
+        with self.assertRaises(KeyError) as ctx:  # missing latitude
+            args = (self.url, self.dev_id, self.key, self.lon, self.name)
+            response = self.client.get("%s/%s/?key=%s&longitude=%s&name=%s" %
```

Comment:
1. Use the [`reverse`](https://docs.djangoproject.com/en/1.11/ref/urlresolvers/) functionality instead of hardcoding the url structure.
2. Instead of passing `?key1=value1&key2=value2` explcitly pass a dictinary as the optional `data` argument:
```
client.get( url, param,   data = {"key" : key_variable, "longitude" : ...})
```